src/nevermined/Nfts.ts

```diff
@@ -512,4 +513,102 @@ export class Nfts extends Instantiable {
             throw Error(`Error listing ${ddo.shortId()} on secondary markets`)
         }
     }
+
+    public async buySecondaryMarketNft(
+        buyer: string,
+        seller: string,
+        nftPrice: number,
+        numOfNfts: number = 1,
+        ddo: DDO
+    ) {
+        const buyersAccount = new Account(buyer)
+        const sellersAccount = new Account(seller)
+        const { token } = this.nevermined.keeper
+
+        const agreementId = utils.generateId()
+        const assetRewards = new AssetRewards(new Map([[seller, nftPrice]]))
+
+        const scale = 10 ** (await token.decimals())
+        await buyersAccount.requestTokens(nftPrice / scale)
+
+        let receipt = await this.nevermined.agreements.conditions.lockPayment(
+            agreementId,
+            ddo.id,
+            assetRewards.getAmounts(),
+            assetRewards.getReceivers(),
+            token.getAddress(),
+            buyersAccount
+        )
+
+        if (!receipt) throw new Error('Transaction Failed.')
+
+        receipt = await this.nevermined.agreements.conditions.transferNft(
+            agreementId,
+            ddo,
+            assetRewards.getAmounts(),
+            assetRewards.getReceivers(),
```

Comment:
With the `assetRewards` that you set up would this mean both the `receivers` and `origin` is the `seller?`

Reply:
I have little idea of asset rewards. Will help if you explain me.